x-pack/plugins/reporting/public/management/components/inspect_in_console_button.tsx

```diff
@@ -0,0 +1,160 @@
+import type { LocatorClient } from '@kbn/share-plugin/common/url_service';
+
+import type { Job } from '../../lib/job';
+import type { KibanaContext } from '../../types';
+
+interface PropsUI {
+  job: Job;
+  csvConfig: ClientConfigType['csv'];
+  searchSourceStart: ISearchStartSearchSource;
+  locators: LocatorClient;
+}
+
+const InspectInConsoleButtonUi: React.FC<PropsUI> = (props) => {
+  const { csvConfig, job, searchSourceStart, locators } = props;
+
+  const { title: jobTitle } = job;
+  const serializedSearchSource = (job.payload as TaskPayloadCSV).searchSource;
+
+  const handleDevToolsLinkClick = useCallback(async () => {
+    const searchSource = await searchSourceStart.create(serializedSearchSource);
+    const index = searchSource.getField('index');
+    if (!index) {
+      throw new Error(`The search must have a reference to an index pattern!`);
+    }
+    const indexPatternTitle = index.getIndexPattern();
+    const pitId = `[ID returned from first request]`; // TODO: translate - user-facing content
+    const queryInfo = getQueryFromCsvJob(searchSource, csvConfig, pitId);
+    const queryUri = compressToEncodedURIComponent(
+      getTextForConsole(jobTitle, indexPatternTitle, queryInfo, csvConfig)
+    );
+    const consoleLocator = locators.get('CONSOLE_APP_LOCATOR');
+    consoleLocator?.navigate({
+      loadFrom: `data:text/plain,${queryUri}`,
+    });
+  }, [searchSourceStart, serializedSearchSource, jobTitle, csvConfig, locators]);
+
+  return (
+    <EuiContextMenuItem
+      data-test-subj="reportInfoFlyoutOpenInConsoleButton"
+      key="download"
+      icon="wrench"
+      onClick={handleDevToolsLinkClick}
+    >
+      {i18n.translate('xpack.reporting.reportInfoFlyout.openInConsole', {
+        defaultMessage: 'Inspect query in Console',
+      })}
+    </EuiContextMenuItem>
+  );
+};
+
+interface Props {
+  job: Job;
+  config: ClientConfigType;
+}
+
+export const InspectInConsoleButton: React.FC<Props> = (props) => {
+  const { config, job } = props;
+  const { services } = useKibana<KibanaContext>();
+  const { application, data, share } = services;
+  const { capabilities } = application;
+  const { locators } = share.url;
+
+  // To show the Console button,
+  // check if job object type is search,
+  // and if both the Dev Tools UI and the Console UI are enabled.
+  const canShowDevTools = job.objectType === 'search' && capabilities.dev_tools.show;
+  if (!canShowDevTools) {
+    return null;
+  }
+
+  return (
+    <InspectInConsoleButtonUi
+      searchSourceStart={data.search.searchSource}
+      locators={locators}
+      job={job}
+      csvConfig={config.csv}
+    />
+  );
+};
+
+const getTextForConsole = (
+  jobTitle: string,
+  indexPattern: string,
+  queryInfo: QueryInspection,
+  csvConfig: ClientConfigType['csv']
+) => {
+  const { requestBody } = queryInfo;
+  const queryAsString = JSON.stringify(requestBody, null, '  ');
+
+  const pitRequest =
```

Comment:
I was wondering if it would make sense to execute that pitRequest and insert its ID in the requestBody above?

Reply:
Got it makes sense 👍